### DISCUSSION TITLE
Lesson 7, Part 6 - advanced deploy script - why did I have to deploy an argument for new FundMe() but Patrick didn't have to in the video?

### DISCUSSION THREAD

**Discussion by 0xwtfwtf** [Cyfrin/foundry-full-course-cu#496] [General]
I'm deploying FundMe.sol on vscode using a script. When Patrick did it in the video, he was able to sandwich "new FundMe()" between vm.startBroadcast() and vm.stopBroadcast() but when I do it, I get the error on the line with new FundMe() that it expects 1 argument but got 0. I figured out from Googling that this was because the constructor in FundMe.sol passes one parameter (the address for priceFeed but FundMe() passes 0 parameters. I fixed it by pasting the priceFeed address into the parentheses, but I'm wondering why I had to do that but Patrick didn't.

Here is my screencap:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/f4159206-d5f9-4ef1-9069-894c2648e7e5)

**Comment by sebghatullahmalikzay** [comment]
in this case, you will create an object of the FundMe after this you can initialize 
write like this:
```
FundMe  fundme = new FundMe();

```

**Reply by 0xwtfwtf** [reply]
When I do that, I get a bunch of new errors about declarations (see screen cap below)...I also tried adding a state variable "fundMe FundMe" but that didn't fix it. I guess if the difference is just that I'm using a newer contract that doesn't hardcode the address, then that is the answer, and I just have to pass an argument equal to the address.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/138697105/1fb48adb-7907-43fc-9904-780a53718b5e)

**Comment by robbiesumner** [comment]
I think Patrick in the beginning Patrick hardcoded the address into the contract, so that you couldn't configure the `priceFeed` address.
You have a newer version of the contract, continue in the course and you'll see that he changes it and later writes a new contract `Helperconfig.sol` to deal with passing this address.

Hope this helps, feel free to ask more questions👍🏼
